### PR TITLE
Return summary with failed pods in WaitForControlledPodsRunning

### PR DESCRIPTION
**Pull request by Argh4k**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR adds summary for WaitForControlledPodsRunning with a list of failed pods to ease the debugging. 

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

**Comment by mborsz** [comment]
Two nits.
Overall looks very good!

/ok-to-test

File: clusterloader2/go.mod [hunk 1]
```diff
@@ -55,6 +55,7 @@ require (
 	k8s.io/client-go v0.22.15
 	k8s.io/component-base v0.22.15
 	k8s.io/component-helpers v0.22.15
+	k8s.io/klog v1.0.0 // indirect
```
[the remark this answers is not in the document]
Comment (Argh4k) [reply clusterloader2/go.mod:58]: I've run `go list -json all` and it looks like it's indirect dependency of pkg/measurement/common/slos.

I did some tests and it looks like this is needed after we merged https://github.com/kubernetes/perf-tests/pull/2174

File: clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go [hunk 2]
```diff
@@ -136,6 +136,46 @@ type waitForControlledPodsRunningMeasurement struct {
 	podsIndexer *measurementutil.ControlledPodsIndexer
 }
 
+type failedPod struct {
+	Namespace    string `json:"namespace"`
```
[the remark this answers is not in the document]
Comment (Argh4k) [reply clusterloader2/pkg/measurement/common/wait_for_controlled_pods.go:140]: I would keep them here, as they clearly state the intent of this struct that is mainly used for packing information before marshaling

**Comment by mborsz** [comment]
/lgtm
/approve

Thanks!

File: clusterloader2/go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (mborsz) [reply clusterloader2/go.mod:58]: Weird... Probably isn't worth debugging

**Pull request merged**